aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/SmokeTestsDecorator.kt

```diff
@@ -0,0 +1,172 @@
+//            val sigv4aRegionSet = node.getArrayMember("sigv4aRegionSet")
+//                .map { a ->
+//                    a.getElementsAs { el ->
+//                        el.expectStringNode().getValue()
+//                    }
+//                }
+//                .orElse(null)
+//            val useAccountIdRouting = node.getBooleanMemberOrDefault("useAccountIdRouting", true)
+            val useDualstack = node.getBooleanMemberOrDefault("useDualstack", false)
+            val useFips = node.getBooleanMemberOrDefault("useFips", false)
+            val uri = node.getStringMemberOrDefault("uri", null)
+            val useAccelerate = node.getBooleanMemberOrDefault("useAccelerate", false)
+//            val useGlobalEndpoint = node.getBooleanMemberOrDefault("useGlobalEndpoint", false)
+            val forcePathStyle = node.getBooleanMemberOrDefault("forcePathStyle", false)
+            val useArnRegion = node.getBooleanMemberOrDefault("useArnRegion", true)
+            val useMultiRegionAccessPoints = node.getBooleanMemberOrDefault("useMultiRegionAccessPoints", true)
+
+            region?.let { writer.rust(".region(crate::config::Region::new(${it.dq()}))") }
+//            sigv4aRegionSet?.let { writer.rust("._($it)") }
+//            useAccountIdRouting?.let { writer.rust("._($it)") }
+            useDualstack?.let { writer.rust(".use_dual_stack($it)") }
+            useFips?.let { writer.rust(".use_fips($it)") }
+            uri?.let { writer.rust(".endpoint_url($it)") }
+            useAccelerate?.let { writer.rust(".accelerate_($it)") }
+//            useGlobalEndpoint?.let { writer.rust(".use_global_endpoint_($it)")}
+            forcePathStyle?.let { writer.rust(".force_path_style_($it)") }
+            useArnRegion?.let { writer.rust(".use_arn_region($it)") }
+            useMultiRegionAccessPoints?.let { writer.rust(".disable_multi_region_access_points(!$it)") }
+        }
+        writer.rust(".build();")
+        writer.dedent()
+    }
+
+    fun renderInput(
+        writer: RustWriter,
+        operationShape: OperationShape,
+        inputShape: StructureShape,
+        data: Optional<ObjectNode>,
+        headers: Map<String, String> = mapOf(),
+        ctx: Ctx = Ctx(),
+    ) {
+        val operationBuilderName =
+            FluentClientGenerator.clientOperationFnName(operationShape, codegenContext.symbolProvider)
+
+        writer.rust("let res = client.$operationBuilderName()")
+        writer.indent()
+        data.orNull()?.let {
+            renderStructureMembers(writer, inputShape, it, headers, ctx)
+        }
+        writer.rust(".send().await;")
+        writer.dedent()
+    }
+
+    fun renderExpectation(
+        writer: RustWriter,
+        expectation: Expectation,
+    ) {
+        if (expectation.isSuccess) {
+            writer.rust("""res.expect("request should succeed");""")
+        } else if (expectation.isFailure) {
+            writer.rust("""res.expect_err("request should fail");""")
```

Comment:
[Smoke test trait ](https://smithy.io/2.0/additional-specs/smoke-tests.html#failureexpectation-struct) error expectation has the expected shape ID, can we assert the error is the correct type?